mountpoint-s3/tests/common/fuse.rs

```diff
@@ -101,7 +120,10 @@ impl TestSession {
 
 impl Drop for TestSession {
     fn drop(&mut self) {
-        // Unmount first by dropping the background session
+        // If the session created with a pre-existing mount (e.g., with `pass_fuse_fd`),
+        // this will unmount it explicitly...
+        let _ = self.mount.take();
```

Comment:
does this call `umount`? If so, I think it is called after the fuse session is shutdown and FD is closed, is it the right order?

Reply:
Yes, `let _` causes `drop` on `Mount` which unmounts the filesystem. The FUSE session - `BackgroundSession` - is dropped next via `self.session.take()`. Although we recently changed to suppress errors during unmount - we were calling `self.session.join()` here [which was also dropping `Mount` first and then waiting for session thread to complete](https://github.com/awslabs/mountpoint-s3/blob/main/vendor/fuser/src/session.rs#L285-L295). 

This ordering sounds correct to me, we call `umount` which closes associated FUSE device (not sure if its also sends a termination signal and wait for a while before closing the device), and the FUSE session reading from associated FUSE device detects it and terminates itself. Though, again, since we removed `self.session.join()` we suppress any error happening on the FUSE session during shutdown, and probably the order doesn't make much difference here.